spec/models/variant_spec.rb

```diff
@@ -0,0 +1,25 @@
+require 'spec_helper'
+
+module Spree
+  RSpec.describe Variant do
+    let(:product) do
+      Product.create!(
+        name: 'globalize',
+        price: 19.99,
+        shipping_category: ShippingCategory.create!(name: 'a')
+      )
+    end
+
+    let!(:variant) do
+      Variant.create!(
+        price: 19.99,
+        product: product
+      )
+    end
+
+    it 'fetches variant from product via translation table' do
+      variants = described_class.includes(:product).ransack(name_cont: 'globalize').result.to_a
+      expect(variants.last).to eq variant
```

Comment:
Yes, I took that part out because it needs Rails 4-2-stable and it looked like a pre-condition check.

I'll take another look at this part.